deployment/docker-swarm/README.md

```diff
@@ -0,0 +1,515 @@
+    - [Upgrade With Black Duck](#upgrade-with-black-duck)
+- [Certificates](#certificates)
+- [Environment Variables](#environment-variables) 
+    - [Edit Environment File](#edit-environment-file)
+    - [Edit the Overrides File](#edit-the-overrides-file)
+    - [Environment Variable Overrides](#environment-variable-overrides)
+    - [Alert Hostname Variable](#alert-hostname-variable)
+    - [Alert Logging Level Variable](#alert-logging-level-variable)
+    - [Email Channel Environment Variables](#email-channel-environment-variables)
+    - [Environment Variable Classifications](#environment-variable-classifications)
+- [Advanced Configuration](#advanced-configuration)
+    - [Changing Server Port](#changing-server-port)
+    - [Changing Memory Settings](#changing-memory-settings)
+
+## Requirements
+
+- A Docker host with at least 2GB of allocatable memory.
+- Administrative access to the docker host machine. 
+- Before installing or upgrading Alert the desired persistent storage volumes must be created for Alert and needs to be either:
+    - Node locked.     
+    - Backed by an NFS volume or a similar mechanism.
+
+## Installing Alert
+Deployment files for Docker Compose are located in the docker-swarm directory of the zip file.
+```
+blackduck-alert-<VERSION>-deployment.zip file.
+```
+- Extract the contents of the ZIP file.
+- For installing with Black Duck the files are located in the *hub* sub-directory.
+- For installing Alert standalone the files are located in the *standalone* sub-directory.
+
+### Standalone Installation
+This section will walk through the instructions to install Alert in a standalone fashion.
+
+#### Overview
+
+1. Create ALERT_ENCRYPTION_PASSWORD secret.
+2. Create ALERT_ENCRYPTION_GLOBAL_SALT secret.
+3. Manage certificates.
+4. Modify environment variables.
+5. Deploy the stack.
+ 
+#### Details 
+This section will walk through each step of the installation procedure.
+
+##### 1. Create ALERT_ENCRYPTION_PASSWORD secret.
+  
+- Create a docker secret containing the encryption password for Alert.
+
+    ```docker secret create <STACK_NAME>_ALERT_ENCRYPTION_PASSWORD <FILE_CONTAINING_PASSWORD>```
+
+- Make sure the alert service is uncommented from the docker-compose.local-overrides.yml file.
+- Uncomment the following from the docker-compose.local-overrides.yml file alert service section.
+    ```
+        alert:
+            secrets:
+                - ALERT_ENCRYPTION_PASSWORD
+    ```
+- Uncomment the following from the secrets section of the docker-compose.local-overrides.yml file.
+    ```
+        secrets:
+            ALERT_ENCRYPTION_PASSWORD:
+              external: true
+              name: "<STACK_NAME>_ALERT_ENCRYPTION_PASSWORD"
+            
+    ```
+##### 2. Create ALERT_ENCRYPTION_GLOBAL_SALT secret.
+
+- Create a docker secret containing the encryption salt for Alert.
+
+    ```docker secret create <STACK_NAME>_ALERT_ENCRYPTION_GLOBAL_SALT - <FILE_CONTAINING_SALT>```
```

Comment:
We should mention they need to replace <STACK_NAME> and <FILE_CONTAINING_SALT> with the values they want to use